### ISSUE TITLE
Stream Buffer doesn't pass through [URL]

### ISSUE THREAD

**Issue by warrentc3** [Threadfin/Threadfin#64]
Stream Buffer doesn't pass through [URL]

**Server (please complete the following information):**
Debian Docker. 
[Threadfin] Version:                1.0 Build: 8
[Threadfin] Database Version:       0.5.0

**To Reproduce**
Enable buffer. Issue happens on both VLC and ffmpeg.
Try to play a channel in VLC client on Windows PC.
Error.

**Expected behavior**
The stream plays.

**Screenshots**
![image](https://user-images.githubusercontent.com/26634421/233525444-f2494281-4327-4461-b9b5-bfa30bfd8bf0.png)

```
2023-04-20 22:09:53 [Threadfin] Buffer:                 true [ffmpeg]
2023-04-20 22:09:53 [Threadfin] Buffer Size:            1024 KB
2023-04-20 22:09:53 [Threadfin] Channel Name:           FYI
2023-04-20 22:09:53 [Threadfin] Client User-Agent:      VLC/3.0.18 LibVLC/3.0.18
2023-04-20 22:09:53 [DEBUG] Create virtual filesystem Folder:/tmp/threadfin/2023-04-7AZ1-4P8JGD/M8K0K03DP9S1ZBRV8GIU/
2023-04-20 22:09:53 [DEBUG] Streaming Status:       Stream ID = 0
2023-04-20 22:09:53 [Threadfin] Streaming Status:       Playlist: frndly - Tuner: 1 / 4
2023-04-20 22:09:53 [DEBUG] Create virtual filesystem Folder:/tmp/threadfin/2023-04-7AZ1-4P8JGD/M8K0K03DP9S1ZBRV8GIU/d41d8cd98f00b204e9800998ecf8427e/
2023-04-20 22:09:53 [Threadfin] FFMPEG path:            /usr/bin/ffmpeg
2023-04-20 22:09:53 [Threadfin] Streaming URL:          
2023-04-20 22:09:53 [DEBUG] FFMPEG:                 /usr/bin/ffmpeg [-user_agent Threadfin -hide_banner -loglevel error -i  -c copy -f mpegts pipe:1]
2023-04-20 22:09:53 [Threadfin] FFMPEG:                 Processing data
2023-04-20 22:09:53 [Threadfin] Streaming Status:       Receive data from FFMPEG
2023-04-20 22:09:54 [Threadfin] FFMPEG log:             : No such file or directory
2023-04-20 22:09:54 [DEBUG] Create virtual filesystem Folder:/tmp/threadfin/2023-04-7AZ1-4P8JGD/M8K0K03DP9S1ZBRV8GIU/d41d8cd98f00b204e9800998ecf8427e/
2023-04-20 22:09:54 [Threadfin] FFMPEG path:            /usr/bin/ffmpeg
2023-04-20 22:09:54 [Threadfin] Streaming URL:          
2023-04-20 22:09:54 [DEBUG] FFMPEG:                 /usr/bin/ffmpeg [-user_agent Threadfin -hide_banner -loglevel error -i  -c copy -f mpegts pipe:1]
2023-04-20 22:09:54 [Threadfin] FFMPEG:                 Processing data
2023-04-20 22:09:54 [Threadfin] Streaming Status:       Receive data from FFMPEG
2023-04-20 22:09:54 [Threadfin] FFMPEG log:             : No such file or directory
```

**Desktop (please complete the following information):**
 - OS: Client = Windows 10
 - Browser Chromium Edge
 - Version 112.0.1722.48

**Comment by warrentc3** [comment]
I think it may be that the buffer isn't fully following redirects.
I'm using this container as a source for this issue.
https://github.com/matthuisman/frndlytv-for-channels

*ip addresses replaced with x.x.x.x*
The m3u loaded into Threadfin contains a channel with this URL:  http://x.x.x.x:8094/play/fyi.m3u8
The docker logs for the frndly container shows this in the log when playing the above URL in VLC directly....
The streams from this provider work fine in ChannelsDVR server.

```
channel/live/fyi > https://sr-live5-frndly.akamaized.net/v1/master/840a24ccd7c75076211d060179b3ad0c64fdc6ca/live_v2_nr_ae/8b37fb68894a497d96eea591eecea052/index.m3u8?ads.rating=tvpg&ads.title=storage%255Fwars&ads.contenttype=live&ads.app_store_url=https://play.google.com/store/apps/details?id=com.tvapp.frndlytv&ads.app_bundle=com.tvapp.frndlytv&ads.station=fyi&ads.us_privacy=1YNY&ads.duration=1800000&ads.genre=reality%252Cbusiness%252Cauction%252CDocumentary&ads.did=d36cc1c5-945b-32fa-ae11-d2e87e215a3f&hdnts=st=1682046412~exp=1682068012~acl=!*/v1/master/840a24ccd7c75076211d060179b3ad0c64fdc6ca/live_v2_nr_ae/8b37fb68894a497d96eea591eecea052/*!/payload/yupptvott_43_1139494_SHIELD30X8X4X0_US_x.x.x.x_frndlytv_1_channel_32_-1/*~data=yupptvott_43_1139494_SHIELD30X8X4X0_US_x.x.x.x_frndlytv_1_channel_32_-1~hmac=**TOKEN**&start=1682046060&startTime=1682046060
x.x.x.x - - [21/Apr/2023 03:06:53] "GET /play/fyi.m3u8 HTTP/1.1" 302 -
```

**Comment by warrentc3** [comment]
OK, so it looks like it's not a buffer thing.  The same behavior is happening without the buffer.  Maybe there's some character in the redirect URL that isn't being escaped or something.

```
2023-04-21 03:31:06 [Threadfin] Buffer:                 false [-]
2023-04-21 03:31:06 [Threadfin] Channel Name:           FYI
2023-04-21 03:31:06 [Threadfin] Client User-Agent:      VLC/3.0.18 LibVLC/3.0.18
2023-04-21 03:31:06 [Threadfin] Streaming URL:          
2023-04-21 03:31:06 [Threadfin] Streaming Info:         URL was passed to the client.
2023-04-21 03:31:06 [Threadfin] Streaming Info:         Threadfin is no longer involved, the client connects directly to the streaming server.
2023-04-21 03:31:06 [Threadfin] [ERROR] Streaming error (Steaming URL could not be found in any playlist) - EC: 1203
2023-04-21 03:31:06 [Threadfin] [ERROR] Streaming error (Steaming URL could not be found in any playlist) - EC: 1203
```

**Comment by warrentc3** [comment]
> 

100% sure.  The same URL plays in VLC that ghosts in Threadfin.

**Comment by Speedyhome72** [comment]
URL Playlist from tvheadend not working ! in Mapping no Channel Names and no URL's 
the same error in log we will playing the channel

**Comment by N3m3515** [comment]
i have exactly the same problem. i am using Version 1.0 (16). Not in Docker

**Comment by bobey6** [comment]
I had this same issue when trying to use the latest version of ffmpeg 6.0 but it works fine with ffmpeg 4.4.2. I changed nothing but the binary and the same streams started working. Looking at the logs, I don't see any URL when using ffmpeg 6.0 but it looks as expected with ffmpeg 4.4

2023/10/01 10:02:34 [Threadfin] FFMPEG path:            /opt/bin/ffmpeg6.0
2023/10/01 10:02:34 [Threadfin] Streaming URL:          
2023/10/01 10:02:34 [Threadfin] FFMPEG:                 Processing data
2023/10/01 10:02:34 [Threadfin] Streaming Status:       Receive data from FFMPEG
2023/10/01 10:02:34 [Threadfin] FFMPEG log:             : No such file or directory


2023/10/01 10:13:09 [Threadfin] FFMPEG path:            /usr/bin/ffmpeg4.4
2023/10/01 10:13:09 [Threadfin] Streaming URL:          http://stream.com
2023/10/01 10:13:09 [Threadfin] FFMPEG:                 Processing data
2023/10/01 10:13:09 [Threadfin] Streaming Status:       Receive data from FFMPEG

**Comment by Fyb3roptik** [comment]
Fixed in 1.0.20
